src/main/kotlin/io/github/gradlenexus/publishplugin/RetrieveStagingProfile.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin
+
+import io.github.gradlenexus.publishplugin.internal.NexusClient
+import org.gradle.api.GradleException
+import org.gradle.api.Incubating
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.property
+import javax.inject.Inject
+
+/**
+ * Diagnostic task for retrieving the [NexusRepository.stagingProfileId] for the [packageGroup] from the provided [NexusRepository] and logging it
+ */
+@Incubating
```

Comment:
@szpak Not sure if we should use this annotation. WDYT?

Reply:
Personally, I like to use `@Incubating` for the new tasks/features to gather feedback and possibly have a clear way to fix spotted issues even in a non-backward compatible way. However, you don't like the "beta" state of that feature or just a Gradle annotation?

Reply:
Ok, fine with me but we should have some common understanding when to do this. IIUC you'd like to do this for all new features. Is that correct?

Reply:
It would be safe, but maybe it would be to defensive, as we - in general - do not expose API to others (at least, it is not the main goal of that extension). How does it look like in JUnit? Are you satisfied with the strategy adopted there?